### PR TITLE
cicd: cleanup CodeQL workflow

### PR DESCRIPTION
**Pull request by RobPasMue**
As title says. Pinging @AlejandroFernandezLuces for review

### CODE REVIEW

File: .github/workflows/codeql.yml [hunk 1]
```diff
@@ -39,36 +22,23 @@ jobs:
...
 
     steps:
     - name: Checkout repository
       uses: actions/checkout@v4
 
-    # Initializes the CodeQL tools for scanning.
     - name: Initialize CodeQL
       uses: github/codeql-action/init@v2
       with:
         languages: ${{ matrix.language }}
```
Comment (AlejandroFernandezLuces) [inline .github/workflows/codeql.yml:33]: We are only analyzing python in this repo, why use the matrix?

**Review by AlejandroFernandezLuces — APPROVED** [review]
Minor comment 🙂

File: .github/workflows/codeql.yml (hunk 1 above)
Comment (RobPasMue) [reply .github/workflows/codeql.yml:33]: Me being lazy 😄 let me remove the matrix

**Review by AlejandroFernandezLuces — APPROVED** [review]
Looking good!

**Pull request merged**